test/unit/org/apache/cassandra/cql3/restrictions/ClusteringElementsTest.java

```diff
@@ -0,0 +1,602 @@
+                assertTrue(rangeSet.contains(twoFiveFive));
+            }
+        }
+    }
+
+    @Test
+    public void testGreaterThanWithThreeColumns()
+    {
+        for (List<ColumnMetadata> columns : asList(newColumns(ASC, ASC, ASC),
+                                                   newColumns(ASC, ASC, DESC),
+                                                   newColumns(DESC, DESC, ASC),
+                                                   newColumns(DESC, DESC, DESC),
+                                                   newColumns(ASC, DESC, ASC),
+                                                   newColumns(ASC, DESC, DESC),
+                                                   newColumns(DESC, ASC, ASC),
+                                                   newColumns(DESC, ASC, DESC)))
+        {
+            ClusteringElements zeroZeroZero = elements(columns, 0, 0, 0);
+            ClusteringElements oneZeroOne = elements(columns, 1, 0, 1);
+            ClusteringElements oneThreeZero = elements(columns, 1, 3, 0);
+            ClusteringElements oneThreeOne = elements(columns, 1, 3, 1);
+            ClusteringElements oneThreeFive = elements(columns, 1, 3, 5);
+            ClusteringElements oneFiveOne = elements(columns, 1, 5, 1);
+            ClusteringElements twoFiveFive = elements(columns, 2, 5, 5);
+
+            RangeSet<ClusteringElements> rangeSet = ClusteringElements.greaterThan(oneThreeOne);
+
+            assertFalse(rangeSet.contains(zeroZeroZero));
+            assertFalse(rangeSet.contains(oneZeroOne));
+            assertFalse(rangeSet.contains(oneThreeZero));
+            assertFalse(rangeSet.contains(oneThreeOne));
+            assertTrue(rangeSet.contains(oneThreeFive));
+            assertTrue(rangeSet.contains(oneFiveOne));
+            assertTrue(rangeSet.contains(twoFiveFive));
+        }
+    }
+
+    private static ClusteringElements elements(ColumnMetadata column, int value)
+    {
+        return ClusteringElements.of(column, bytes(value));
+    }
+
+    private static ClusteringElements elements(List<ColumnMetadata> columns, int... values)
+    {
+        return ClusteringElements.of(columns, Arrays.stream(values).mapToObj(ByteUtils::bytes).collect(Collectors.toList()));
+    }
+
+    private static ColumnMetadata newColumn(AbstractType<?> type)
+    {
+        return newColumn(type, 0);
+    }
+
+    private static ColumnMetadata newColumn(AbstractType<?> type, int position)
+    {
+        return ColumnMetadata.clusteringColumn("ks", "tbl", "c" + position, type, position);
+    }
+
+    private static List<ColumnMetadata> newColumns(AbstractType<?>... types)
+    {
+        List<ColumnMetadata> columns = new ArrayList<>(types.length);
+        for (int i = 0, m = types.length; i < m; i++)
+        {
+            columns.add(newColumn(types[i], i));
+        }
+        return columns;
+    }
+
+    private static List<ColumnMetadata> extend(List<ColumnMetadata> columns, AbstractType<?> type)
```

Comment:
```suggestion
    private static List<ColumnMetadata> addColumns(List<ColumnMetadata> columns, AbstractType<?> type)
```
We have already extend method in ClusteringElements, so extend here sounds confusing as we do not call that one.

Reply:
I also noticed we do not test `ClusteringElements#extend()`